File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
    int n = nums.size();
    if (n == 0) return 0;
    sort(nums.begin(), nums.end());
    int smaller = INT_MIN;
    int cnt = 0;
    int longer = 1;
    for (int i = 0; i < n; i++) {
        if (nums[i] - 1 == smaller) {
            cnt += 1;
            smaller = nums[i];
        }
        else if (nums[i] != smaller) {
            cnt = 1;
            smaller = nums[i];
        }
        longer = max(longer, cnt);
    }
    return longer; }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        seen =set(nums)
        maxi=0
        for num in seen:
            if (num-1) not in seen:
                l=1
                while(num+l) in seen:
                    l+=1
                maxi=max(l,maxi)
        return maxi
